Assets/_Scripts/CurveController.cs

```diff
@@ -1,96 +1,138 @@
 using System.Collections.Generic;
 using UnityEngine;
-using UnityEngine.Splines;
 
 public class CurveController : MonoBehaviour
 {
     [SerializeField]
-    GameObject marker;
+    private GameObject marker;
 
-    [SerializeField]
-    InputManager inputManager;
-
-    [SerializeField]
-    MarkerSelection markerSelection;
-
-    [SerializeField]
-    SplineController splineController;
+    //[SerializeField] private InputManager inputManager;
 
     [SerializeField]
-    GhostKnotController ghostKnotController;
+    private GhostKnotController ghostKnotController;
 
     [SerializeField]
-    GameObject UI_FrameInputWindow;
+    private GameObject UI_FrameInputWindow;
 
-    private bool leftMouseButtonIsPressed = false;
-    private GameObject selectedMarker;
-    private GameObject selectedObject;
     public List<GameObject> markerList = new List<GameObject>();
```

Comment:
We do have a class that represents the Markers called `MarkerEntity`. We should always use the MarkerEntity in our data structures such as this one (instead of storing the GameObjects). The MarkerEntity is basically a mono behaviour that also contains the GameObject + much more information about the marker.